sparse/src/KokkosSparse_Utils.hpp

```diff
@@ -2330,6 +2330,128 @@ void validateCrsMatrix(int m, int n, const Rowmap &rowmapIn,
   }
 }
 
+template <typename crsMat_t>
+void kk_extract_diagonal_blocks_crsmatrix_sequential(
+    const crsMat_t &A, std::vector<crsMat_t> &DiagBlk_v) {
+  using row_map_type            = typename crsMat_t::row_map_type;
+  using entries_type            = typename crsMat_t::index_type;
+  using values_type             = typename crsMat_t::values_type;
+  using row_map_hostmirror_type = typename row_map_type::HostMirror;
+  using entries_hostmirror_type = typename entries_type::HostMirror;
+  using values_hostmirror_type  = typename values_type::HostMirror;
+  using int_view1d_type =
+      Kokkos::View<int *, Kokkos::LayoutLeft, Kokkos::HostSpace>;
+
+  using graph_t          = typename crsMat_t::StaticCrsGraphType;
+  using out_row_map_type = typename graph_t::row_map_type::non_const_type;
+  using out_entries_type = typename graph_t::entries_type::non_const_type;
+  using out_values_type  = typename crsMat_t::values_type::non_const_type;
+  using out_row_map_hostmirror_type = typename out_row_map_type::HostMirror;
+  using out_entries_hostmirror_type = typename out_entries_type::HostMirror;
+  using out_values_hostmirror_type  = typename out_values_type::HostMirror;
+
+  row_map_type A_row_map = A.graph.row_map;
+  entries_type A_entries = A.graph.entries;
+  values_type A_values   = A.values;
+
+  row_map_hostmirror_type A_row_map_h =
+      Kokkos::create_mirror_view_and_copy(Kokkos::HostSpace(), A_row_map);
+  entries_hostmirror_type A_entries_h =
+      Kokkos::create_mirror_view_and_copy(Kokkos::HostSpace(), A_entries);
+  values_hostmirror_type A_values_h =
+      Kokkos::create_mirror_view_and_copy(Kokkos::HostSpace(), A_values);
+
+  int A_nrows  = static_cast<int>(A_row_map.extent(0)) - 1;
```

Comment:
Another question, what should happen if a user passes in a rectangular matrix instead of square? Should there be a check and a throw for that case?

Reply:
@ndellingwood I added a check and throw for square matrix